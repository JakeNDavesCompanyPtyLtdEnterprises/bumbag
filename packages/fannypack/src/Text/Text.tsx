import * as React from 'react';
import * as PropTypes from 'prop-types';
import { BoxProps as ReakitBoxProps } from 'reakit/ts/Box/Box';

import _Text from './styled';

export type LocalTextProps = {
  className?: string;
  children?: React.ReactNode;
  color?: string;
};
export type TextProps = ReakitBoxProps & LocalTextProps;

export const Text: React.FunctionComponent<LocalTextProps> = ({ children, ...props }) => (
  <_Text use="span" {...props}>
    {children}
  </_Text>
);

Text.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  color: PropTypes.string
};
Text.defaultProps = {
  className: undefined,
  children: undefined,
  color: undefined
};

// @ts-ignore
const C: React.FunctionComponent<TextProps> = Text;
export default C;
