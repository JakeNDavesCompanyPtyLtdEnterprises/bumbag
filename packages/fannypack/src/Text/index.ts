export { default as Text } from './Text';
export { default } from './Text';
