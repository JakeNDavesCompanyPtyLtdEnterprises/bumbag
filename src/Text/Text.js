import React, { type Node } from 'react';

import type { Palette } from '../types';
import _Text from './styled';

type Props = {
  use?: any,
  className?: string,
  children: Node,
  color?: Palette
};

const Text = ({ className, children, ...props }: Props) => (
  <_Text className={className} {...props}>
    {children}
  </_Text>
);

Text.defaultProps = {
  use: 'span',
  className: undefined,
  color: undefined
};

export default Text;
