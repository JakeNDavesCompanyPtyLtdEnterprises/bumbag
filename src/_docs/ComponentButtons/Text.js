import React from 'react';

const SvgComponent = props => (
  <svg width={78} height={4} fill="none" {...props}>
    <path d="M78 0H0v4h78V0z" fill="#3926A5" />
  </svg>
);

export default SvgComponent;
