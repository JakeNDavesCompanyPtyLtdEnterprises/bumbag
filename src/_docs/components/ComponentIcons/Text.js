import React from 'react';

const SvgComponent = props => (
  <svg width={78} height={4} fill="none" {...props}>
    <path d="M78 0H0v4h78V0z" fill="#444bc9" />
  </svg>
);

export default SvgComponent;
