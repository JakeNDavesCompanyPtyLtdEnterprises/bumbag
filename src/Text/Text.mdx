---
name: Text
route: /typography/text
menu: Typography
---

import { Playground, PropsTable } from 'docz';
import { Box } from '../primitives/index';
import Text from './index';

# Text

## Import

`import { Text } from 'fannypack'`

## Basic Usage

The `<Text>` component renders a `<span>` with basic reset styles by default.

<Playground>
  <Text>Hello world!</Text>
</Playground>

> Remember, you can also specify different colors with the `color` prop.

<Playground>
  <Text color="success">Hello world!</Text><br />
  <Text color="danger">Hello world!</Text>
</Playground>

## Other elements

<Playground>
  <Text use="strong">Bold</Text><br />
  <Text use="i">Italic</Text><br />
  <Text use="u">Underline</Text><br />
  <Text use="abbr">I18N</Text><br />
  <Text use="cite">Citation</Text><br />
  <Text use="del">Deleted</Text><br />
  <Text use="em">Emphasis</Text><br />
  <Text use="ins">Inserted</Text><br />
  <Text use="kbd">Ctrl + C</Text><br />
  <Text use="mark">Highlighted</Text><br />
  <Text use="s">Strikethrough</Text><br />
  <Text use="samp">Sample</Text><br />
  Text<Text use="sub">sub</Text><br />
  Text<Text use="sup">sup</Text>
</Playground>

## Props

<PropsTable of={Text} />

## Theming

### Schema

```jsx
{
  base: string | Object
}
```
