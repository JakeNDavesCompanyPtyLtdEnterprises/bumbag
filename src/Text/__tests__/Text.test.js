import React from 'react';
import render from '../../_utils/tests/render';
import Text from '../Text';
import 'jest-styled-components';

it('renders correctly for a basic text', () => {
  const { container } = render(<Text>test</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for text with a color', () => {
  const { container } = render(<Text color="primary">test</Text>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders correctly for text with a another element', () => {
  const { container } = render(
    <Text use="strong" color="primary">
      test
    </Text>
  );
  expect(container.firstChild).toMatchSnapshot();
});
