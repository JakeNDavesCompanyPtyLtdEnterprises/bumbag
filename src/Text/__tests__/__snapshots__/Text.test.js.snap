// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a basic text 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

abbr.c0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c0 {
  font-style: italic;
}

kbd.c0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c0 {
  font-weight: bold;
}

sub.c0,
sup.c0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c0,
u.c0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c0,
cite.c0,
var.c0,
em.c0 {
  font-style: italic;
}

kbd.c0,
pre.c0,
samp.c0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c0 .c2 {
  top: 0.1em;
}

<span
  class="c0 c1"
>
  test
</span>
`;

exports[`renders correctly for text with a another element 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #444bc9 !important;
}

abbr.c0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c0 {
  font-style: italic;
}

kbd.c0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c0 {
  font-weight: bold;
}

sub.c0,
sup.c0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c0,
u.c0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c0,
cite.c0,
var.c0,
em.c0 {
  font-style: italic;
}

kbd.c0,
pre.c0,
samp.c0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c0 .c2 {
  top: 0.1em;
}

<strong
  class="c0 c1"
  color="primary"
>
  test
</strong>
`;

exports[`renders correctly for text with a color 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
  color: #444bc9 !important;
}

abbr.c0 {
  border-bottom: 1px dotted;
  cursor: help;
  -webkit-text-decoration: none;
  text-decoration: none;
}

i.c0 {
  font-style: italic;
}

kbd.c0 {
  background: #454d5d;
  border-radius: 0.1rem;
  color: #fff;
  fill: #fff;
  padding: 0.1rem 0.2rem;
}

mark.c0 {
  background: #ffe9b3;
  border-radius: 0.1rem;
  padding: 0.1rem 0.2rem;
}

strong.c0 {
  font-weight: bold;
}

sub.c0,
sup.c0 {
  font-size: 0.8em;
  vertical-align: baseline;
}

ins.c0,
u.c0 {
  -webkit-text-decoration-skip: ink edges;
  text-decoration-skip: ink edges;
}

i.c0,
cite.c0,
var.c0,
em.c0 {
  font-style: italic;
}

kbd.c0,
pre.c0,
samp.c0 {
  font-family: 'SF Mono','Segoe UI Mono','Roboto Mono',Menlo,Courier,monospace;
}

.c0 .c2 {
  top: 0.1em;
}

<span
  class="c0 c1"
  color="primary"
>
  test
</span>
`;
